hack/test/trap

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+#
+# This test ensures that the functions in 'os::util::trap' function as expected
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/cmd_util.sh"
+source "${OS_ROOT}/hack/lib/util/trap.sh"
+cd "${OS_ROOT}"
+
+# basic functionality tests
+some_funcname="func"
+VERBOSE=true os::cmd::expect_success "os::util::trap::add ${some_funcname} ERR"
+os::cmd::expect_success_and_text "echo ${REGISTERED_TRAPS_TO_SIGNALS[${some_funcname}]-}" "ERR"
```

Comment:
New issue: the associative arrays are once again completely empty at this point. They retain the values written to them until the last line of `os::util::trap::add`, then go blank.

Adding, accessing, and removing elements inside of the test script explicitly works OK. I wrote a gist to try to get this to occur in a smaller system but I can't. Link to [Gist](https://gist.github.com/stevekuznetsov/11ec05a03253f5069fab).